OsmAnd-java/src/main/java/net/osmand/router/RouteResultPreparation.java

```diff
@@ -176,11 +176,54 @@ private boolean segmentLineBelongsToPolygon(CombineAreaRoutePoint p, CombineArea
 		combineWayPointsForAreaRouting(ctx, result);
 		validateAllPointsConnected(result);
 		splitRoadsAndAttachRoadSegments(ctx, result, recalculation);
+		for (int i = 0; i < result.size(); i++) {
+			filterMinorStops(result.get(i));
+		}
 		calculateTimeSpeed(ctx, result);
-		
 		prepareTurnResults(ctx, result);
 		return result;
 	}
+	
+	public RouteSegmentResult filterMinorStops(RouteSegmentResult seg) {
+		int startPoint = seg.getStartPointIndex();
+		int endPoint = seg.getEndPointIndex();
+		int start;
+		int end;
+
+		if (startPoint < endPoint) {
+			start = startPoint;
+			end = endPoint;
+		} else {
+			start = endPoint;
+			end = startPoint;
+		}
+
+		List<Integer> stops = new ArrayList<>();
```

Comment:
Better not create object if it's not going to be used i.e. 
List<Integer> stops = null;
....
if(stops == null) {
... // create
}

if (stops != null) { 
// remove and check

}